src/systems/systems/src/hydraulic/brake_circuit.rs

```diff
@@ -252,6 +259,54 @@ impl BrakeCircuit {
 
         self.accumulator_fluid_pressure_sensor_filter
             .update(context.delta(), actual_pressure_available);
+
+        self.update_failures(context, section);
+    }
+
+    fn update_failures(&mut self, context: &UpdateContext, section: &impl SectionPressure) {
+        if let Some((precharge_failure, accumulator)) = self
+            .accu_gas_precharge_failure
+            .as_ref()
+            .zip(self.accumulator.as_mut())
+        {
+            if precharge_failure.is_active() {
+                let current_pre_charge_pressure_in_accumulator =
+                    accumulator.gas_precharge_pressure();
+
+                let new_pressure_after_leak = (current_pre_charge_pressure_in_accumulator
+                    - Pressure::new::<psi>(
+                        context.delta_as_secs_f64()
+                            * Self::ACCUMULATOR_GAS_FAILURE_LEAKING_GRADIENT_PSI_PER_S,
+                    ))
+                .max(Pressure::new::<psi>(
+                    Self::ACCUMULATOR_GAS_FAILURE_MIN_ALLOWED_PRESSURE_PSI,
+                ));
+
+                accumulator.set_gas_precharge_pressure(new_pressure_after_leak);
+            } else if self.accu_gas_precharge_failure_active_previous_state
+                && !precharge_failure.is_active()
+            {
```

Comment:
```suggestion
            } else if self.accu_gas_precharge_failure_active_previous_state {
```
The else branch is only executed when the failure is not active so it's not necessary to test that again.